### ISSUE TITLE
Configure Pinpoint SMS sender

### ISSUE THREAD

**Issue by willeybryan** [cds-snc/notification-api#732]
**Story**
When sending SMS from my Notify service to the public I want it to come from the same number(s) every time so that we can build a trusting communication relationship

**What**
- Configure pinpoint for one dedicated long code
- Test it to see if it works for sending 1 sms at a time
- Test to see what happens if we through 10 at it or something

**Why?**
- Government comms people don't trust SNS's American numbers and don't want to send from that

**Done When?**
- We can safely offer long codes to a service for their use. 

**Notes**
Pinpoint https://docs.aws.amazon.com/pinpoint/latest/userguide/channels-sms-countries.html

**Comment by timarney** [comment]
Docs
https://docs.aws.amazon.com/pinpoint/latest/developerguide/send-messages-sms.html

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/pinpoint.html

**Comment by timarney** [comment]
Based on the docs above we'll be looking to adjust the current SES client send to use PinPoint

> I'm pretty sure you can mirror the ses client and tweak a few things @obrien-j

**Comment by willeybryan** [comment]
Notes for remembering
- If a platform admin bumps up Pinpoint to the top priority, that would stop sms from being delivered for any service without an inbound sms assigned.
- We are still using SNS for sending sms, except when a service has an "inbound sms" assigned to it. For that we use Pinpoint.
- if we don't assign or create any longcodes, there won't be any inbound SMS numbers available to assign to services. All sms will be sent through SNS as before
- "Does every service need a long code?" No - until we have rate limiting in place, we should not assign inbound sms to any services. Otherwise we could exceed the rate limits enforced by telcos and get our longcodes blocked. Once that code is in place we could start with 1 service and see how that goes before assigning more.
- STOP/HELP messages are configurable in the Pinpoint dashboard - it looks like they will be the same messages across all our long codes for now

**Comment by willeybryan** [comment]
Rate limiting seems like the continuation of this ticket.

**Comment by timarney** [comment]
> Rate limiting seems like the continuation of this ticket.

Correct

**Comment by ClementineHahn** [comment]
Hi @smcmurtry @timarney , what would you say is left to do to consider this done ? 
Because from my understanding/memories there is nothing left but well im not sure i have the best understanding of it ;)  Thank you !

**Comment by timarney** [comment]
Another card could be opened for "Rate limiting"

https://github.com/cds-snc/notification-api/issues/732#issuecomment-621751346

**Comment by smcmurtry** [comment]
Bryan opened this issue for Rate limiting: #807

Will close this now since the actual pinpoint implementation is done.
